xmpp-vala/src/core/xmpp_stream.vala

```diff
@@ -57,10 +58,14 @@ public class XmppStream {
             }
             IOStream? stream = null;
             if (best_provider != null) {
-                stream = yield best_provider.connect(this);
+                stream = yield best_provider.connect(this, out remote_hostname);
             }
             if (stream == null) {
-                stream = yield (new SocketClient()).connect_async(new NetworkService("xmpp-client", "tcp", this.remote_name.to_string()));
+                GLibFixes.Resolver resolver = GLibFixes.Resolver.get_default();
```

Comment:
This originally was the fallback case for when there is no SRV record. I guess your change will break the fallback and make SRV records required.

Reply:
My understanding is that the original code would only work with SRV as well, based on reading the documentation for NetworkService: https://valadoc.org/gio-2.0/GLib.NetworkService.html

Wouldn't this need to use NetworkAddress rather than NetworkService to use a plain hostname rather than a SRV record?

Reply:
I'm marking this as resolved since I've reverted this part of the code - whether this is doing the right thing is orthogonal.